### DISCUSSION TITLE
Can we add the option (or by default) so that empty folders get deleted after running the rip? (raw/transcode/media/movies)

### DISCUSSION THREAD

**Discussion by noideas4thisname** [automatic-ripping-machine/automatic-ripping-machine#871] [Ideas]
As it stands, ARM is creating a ton of empty folders and it is bothering me quite a bit in terms of organizing the final destination. I think it's fine for the temporary folders to be created and utilized as part of the ripping process, but can we add the ability in the Delete Raw setting, Skip Transcode setting, and other related settings to remove the folder itself if it is empty?

As it stands the docker container spits out 5 folders where 4 of them are useless in my setup:
![image](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/2354213/1d3e95ba-48f0-4c8a-a9c8-e508425bc606)

A couple of notes:
- I have Skip Transcode on. In this case, I'd prefer the transcode folder doesn't exist.
- I have Delete Raw on. I'd prefer this get deleted as part of the cleanup process as well.
- No idea why there is a movies folder being created. Movies themselves only end up in the completed folder.
- I assume the media folder is for disc backups, but again - never been used and I don't see why it wouldn't also go in completed.

Anyways - thought I'd start a discussion to see thoughts on this. It may be something I could look at implementing later if I get spare time to familiarize myself with the sections taking these actions, but checking if its easy/desired to add by those more familiar.

**Comment by drlauridsen** [comment]
One question.. if you have skip transcode on & Delete Raw on - dont you loose all the files that have just been extracted from the dvd - since they are only placed in the RAW folder?

**Reply by noideas4thisname** [reply]
No. You get completed .mkv files (full quality) in the completed folder that have all the content you want. You don't need Raw files because you arent transcoding.

**Reply by noideas4thisname** [reply]
I should add that the raw file path is used as a temporary ripping location while it is running, but it then moves it to completed. Hence why I'd like the raw folder to go away once its temporary purpose is completed.

**Comment by drlauridsen** [comment]
Thanks, I just read somewhere, that I had to turn off delete raw, when not using transcoding, but just realized that the files are indeed,  as you say, in the completed folder, and with correct naming.

I then completely agree with you in regards to cleaning up after finishing.. at least as a user setting

**Comment by microtechno9000** [comment]
At the moment the ripper code is not particularly clean and easily modified.

The devs are working on slowly rewriting the ripper side of ARM to allow easier management and thus allow for additional features.

Things like this request could then be implemented. That or I think there is a separate request to have post job actions, like moving files to a NAS etc.

If you would like to raise an issue, or feature request we can tag as an enhancement for future work.

Note there are only three main devs, so progress is slow.
